Quadriga/src/main/java/edu/asu/spring/quadriga/web/search/TextConceptSearchController.java

```diff
@@ -53,84 +56,91 @@
-
-                ITransformedNetwork transformedNetwork = transformationManager
-                        .getSearchTransformedNetworkMultipleProjects(projectIds, conceptUri, INetworkStatus.APPROVED);
-
-                String json = null;
-                if (transformedNetwork != null) {
-                    json = jsonCreator.getJson(transformedNetwork.getNodes(), transformedNetwork.getLinks());
-                }
+                String results = qStoreConnector.searchNodesByConcept(conceptId);
+                if (results != null && !results.isEmpty()) {
+                    ElementEventsType events = marshallingService.unMarshalXmlToElementEventsType(results);
+                    List<CreationEvent> eventList = events.getRelationEventOrAppellationEvent();
+                    if (concept.getWordnetIds() != null && !concept.getWordnetIds().isEmpty()) {
+                        results = qStoreConnector.searchNodesByConcept(concept.getWordnetIds().get(0));
+                        events = marshallingService.unMarshalXmlToElementEventsType(results);
+                        eventList.addAll(events.getRelationEventOrAppellationEvent());
+                    }
 
-                if (transformedNetwork == null || transformedNetwork.getNodes().size() == 0) {
-                    model.addAttribute("isNetworkEmpty", true);
+                    Set<String> references = new HashSet<String>();
+                    for (CreationEvent event : eventList) {
+                        String sourceRef = event.getSourceReference();
+                        references.add(sourceRef);
+                    }
+                    referencesList.add(references);
                 }
+                concepts.add(concept);
+                conceptUriSearchList.add(conceptUri);
 
-                model.addAttribute("jsonstring", json);
-                
-            }            
-            
-            model.addAttribute("concept", concept);
+            }
         }
 
+        if (referencesList.size() >= 1) {
+            Set<String> resultSet = referencesList.get(0);
+            for (int i = 1; i < referencesList.size(); i++) {
+                resultSet = Sets.intersection(resultSet, referencesList.get(i));
+            }
+            Iterator<String> it = resultSet.iterator();
+            while (it.hasNext()) {
+                String sourceRef = it.next();
+                ITextFile txtFile = textFileManager.getTextFileByUri(sourceRef);
+                if (txtFile == null || txtFile.getAccessibility() == ETextAccessibility.PRIVATE) {
+                    handles.add(sourceRef);
+                } else {
+                    if (txtFile.getAccessibility() == ETextAccessibility.PUBLIC) {
+                        texts.add(txtFile);
+                        textFileManager.loadFile(txtFile);
+                        txtFile.setSnippetLength(40);
+                    }
+                }
+            }
+        }
```

Comment:
this is only getting all texts that contain all concepts, not all texts that contain all concepts in one statement, doesn't it? your network will be different from your texts.